src/store/tests/Document/Loader/RssLoaderTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Document\Loader;
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+use Symfony\AI\Store\Document\Loader\RssLoader;
+use Symfony\AI\Store\Document\TextDocument;
+use Symfony\AI\Store\Exception\InvalidArgumentException;
+use Symfony\AI\Store\Exception\RuntimeException;
+use Symfony\Component\HttpClient\MockHttpClient;
+use Symfony\Component\HttpClient\Response\MockResponse;
+
+/**
+ * @author Niklas Grießer <niklas@griesser.me>
+ */
+#[CoversClass(RssLoader::class)]
+final class RssLoaderTest extends TestCase
+{
+    public function testLoadWithNullSource()
+    {
+        $loader = new RssLoader();
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('"Symfony\AI\Store\Document\Loader\RssLoader" requires a URL as source, null given.');
+
+        iterator_to_array($loader->load(null));
+    }
+
+    public function testLoadWithValidRssFeed()
+    {
+        $httpClient = new MockHttpClient([new MockResponse(file_get_contents(__DIR__.'../../fixtures/symfony-feed.xml'))]);
```

Comment:
```suggestion
        $httpClient = new MockHttpClient([MockResponse::fromFile(__DIR__.'../../fixtures/symfony-feed.xml')]);
```
Should be available in 6.4